tests/typechecker/auto_deref.jakt

```diff
@@ -47,8 +47,8 @@ function deref_match(anon foo: &i32) -> i32 {
 }
 
 struct Foo {
-    function const_method(this) => 69
-    function mutable_method(mut this) => 69 + 1
+    function const_method(this) -> i32 => 69i32
```

Comment:
I think we should be able to infer the type of fat arrow functions, as we're immediately returning what they point to

Reply:
The inferred type here is i64 but the function was used as it was i32 in deref_access function, which returns i32. Before we would just silently accept the overflow and wrap the number. Not sure what the correct behavior is...?

```cpp
struct Foo {
    function const_method(this) -> i32 => 69i32
    function mutable_method(mut this) -> i32 => 69i32 + 1i32
}

function deref_access(anon foo: &mut Foo) -> i32 {
    return foo.const_method() + foo.mutable_method()
}
```

Reply:
Actually I was wrong, after my changes the overflow of i32 in checked addition here would cause a runtime panic.

Reply:
Unless, you meant that we only need to say `=> 69i32` instead of `-> i32 => 69i32`, then yes you are right 😅 I wrote the return type out of habit

Reply:
Right, the last one. It should just be `=> 69i32` and infer `i32` as the return type.